string/location.ts

```diff
@@ -0,0 +1,18 @@
+import { TextWithLocation } from "./types.ts";
+
+export function locateRegex(text: string, regex: RegExp): TextWithLocation[] {
```

Comment:
I think what we want is even simpler - just `location(text: string, index: number): TextLocation`. This can then be used in evalCodeBlocks but also anywhere else where we want to translate between an index (as given by regex match.index in this case, but there are other situations where this can be obtained) and a more sophisticated `TextLocation` description.

Reply:
Added `location()`